vavr/src/main/java/io/vavr/collection/LinkedHashMap.java

```diff
@@ -701,14 +701,14 @@ public boolean isSequential() {
 
     @Override
     public LinkedHashMap<K, V> put(K key, V value) {
-        Queue<Tuple2<K, V>> newList = list;
-        HashMap<K, V> newMap = map;
-        if (containsKey(key)) {
-            newList = newList.filter(t -> !Objects.equals(t._1, key));
-            newMap = newMap.remove(key);
+        final Queue<Tuple2<K, V>> newList;
+        final Option<V> currentEntry = get(key);
+        if (currentEntry.isDefined()) {
```

Comment:
maybe we should check that `oldValue.equals(newValue)` is `true` and in this case do nothing (?)